src/app/webui/components/ui/markdown-text.tsx

```
@@ -0,0 +1,60 @@
+
+// Enhanced markdown component with proper emoji support and spacing
+const MarkdownTextImpl = ({ children }: { children: string }) => {
+  return (
+    <div
+      className="prose prose-sm max-w-none dark:prose-invert [&>p]:mb-4 [&>p]:last:mb-0 [&>h1]:text-xl [&>h1]:font-bold [&>h1]:mb-4 [&>h1]:mt-6 [&>h1]:first:mt-0 [&>h2]:text-lg [&>h2]:font-bold [&>h2]:mb-3 [&>h2]:mt-5 [&>h2]:first:mt-0 [&>h3]:text-base [&>h3]:font-bold [&>h3]:mb-2 [&>h3]:mt-4 [&>h3]:first:mt-0 [&>ul]:mb-4 [&>ul]:ml-6 [&>ul]:list-disc [&>ul]:space-y-1 [&>ol]:mb-4 [&>ol]:ml-6 [&>ol]:list-decimal [&>ol]:space-y-1 [&>li]:leading-relaxed [&>blockquote]:border-l-4 [&>blockquote]:border-muted [&>blockquote]:pl-4 [&>blockquote]:my-4 [&>blockquote]:italic [&>blockquote]:text-muted-foreground [&>hr]:my-6 [&>hr]:border-border [&_p]:[white-space:pre-line] [&_p]:[word-break:keep-all] [&_p]:[overflow-wrap:normal] [&_p]:[hyphens:none] [&_li]:[white-space:pre-line] [&_li]:[word-break:keep-all] [&_li]:[overflow-wrap:normal] [&_li]:[hyphens:none]"
+    >
+      <ReactMarkdown 
+        remarkPlugins={[remarkGfm]}
+        components={{
+          code: ({ className, children, ...props }) => {
+            const [copied, setCopied] = useState(false);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)